python-checks/src/test/resources/checks/datetime_constructor_pytz_timezone.py

```diff
@@ -0,0 +1,37 @@
+import datetime
+import pytz
+
+dt = datetime.datetime(2022, 1, 1, tzinfo=pytz.timezone('US/Eastern'))  # Noncompliant
+
+def timezone_in_a_variable():
+    some_timezone = pytz.timezone('US/Eastern')
+    dt = datetime.datetime(2022, 1, 1, tzinfo=some_timezone)  # Noncompliant
+    return dt
+
+some_timezone_1 = pytz.timezone('US/Eastern')
+dt1 = datetime.datetime(2022, 1, 1, tzinfo=some_timezone_1)  # FN because of ReachingDefinitionsAnalysis limitations
```

Comment:
What limitation are you referring to? (no need to add it in the test - this is a legitimate question)

I believe the limitation is that it doesn't actually run on module level code and only does so within functions?

If yes, it's totally fair to document it. However, I feel it would then make sense to place the other calls to `datetime.datetime` below within a function, so that it's clear that this limitation cannot impact those tests (even though it doesn't anyway since no intermediate variable is involved for those tests).